File: Project Οντοκεντρικός Προγραμματισμός/C++ Project/CodeBlocks Project Hotel/src/Booking.cpp
#include <Booking.h>

int Booking::bookingInstanceCounter = 0; //Initialization of the variable "bookingInstanceCounter" at 0, used for ID binding.
string clientName;//Variable used for the name of the client for a Booking object.
int arrivalDay;//Variable used for the arrival day of a Booking object.
int accomodationDays;//Variable used for the accomodation days of a Booking object.
int peopleNum;//Variable used for the number of people of a Booking object.
int bookingID; //Variable used for ID binding.
static int bookingInstanceCounter; //Variable used for ID binding.
Room *room;//Room type pointer.


Booking::Booking(string name, int _arrivalDay, int _accomodationDays, int _peopleNum)//Constructor for a Booking type object.
{
	clientName = name;
	arrivalDay = _arrivalDay;
	accomodationDays = _accomodationDays;
	peopleNum = _peopleNum;
	setBookingID();//Increasing the bookingID variable by each successful booking constructed.
}

Booking::~Booking()//Default destructor for a Booking type object.
{
}

string getClientName()//Getter for the variable clientName.
{
	return clientName;
};

void setClientName(string val)//Setter for the variable clientName.
{
	clientName = val;
}

int getArrivalDay()//Getter for the variable arrivalDay.
{
	return arrivalDay;
}

void setArrivalDay(int val)//Setter for the variable arrivalDay.
{
	arrivalDay = val;
}

int getAccomodationDays()//Getter for the variable accomodationDays.
{
	return accomodationDays;
}

void setAccomodationDays(int val)//Setter for the variable accomodationDays.
{
	accomodationDays = val;
}

int getPeopleNum()//Getter for the variable peopleNum.
{
	return peopleNum;
}

void setPeopleNum(int val)//Setter for the variable peopleNum.
{
	peopleNum = val;
}

void setBookingID()//Setter for the variable bookingID.
{
	bookingID = ++bookingInstanceCounter;
}

int getBookingID()//Getter for the variable bookingID.
{
	return bookingID;
}

void setRoom(Room *a)//Setter for the Room type pointer.
{
	room = a;
}

Room* getRoom()//Getter for the Room type pointer.
{
	return room;
}

File: Project Οντοκεντρικός Προγραμματισμός/Java Project/Hotel Project, Swing/Booking.java
public class Booking
{
    private String clientName="No name";
    private static int instanceCounter;//Μεταβλητή για την αντιστοίχιση του ID.
    private int bookingID;             //Μεταβλητή για την αντιστοίχιση του ID.
    private int arrivalDate;           //Μεταβλητή για την μέρα άφιξης.
    private int accomodationDays;      //Μεταβλητή για τις μέρες διαμονής.
    private int peopleNum;             //Μεταβλητή για τον αριθμό των ατόμων.
    private Room room;                 //Μεταβλητή τύπου δωμάτιο.
    //Setters
    public void setBookingID() 
    {
        instanceCounter++;
        bookingID = instanceCounter;
    }
    public void setClientName(String clientName)
    {
        this.clientName=clientName;
    }
    public void setArrivalDate(int arrivalDate) 
    {
        this.arrivalDate=arrivalDate;
    }
    public void setAccomodationDays(int accomodationDays)
    {
        this.accomodationDays=accomodationDays;
    }
    public void setPeopleNum(int peopleNum)
    {
        this.peopleNum=peopleNum;
    }
    public void setRoom(Room roomToGetAssigned ) //Μέθοδος η οποία αναθέτει ένα αντικείμενο τύπου δωμάτιο στην μεταβλητή 'δωμάτιο' αυτής της κλάσης.
    {
        room=roomToGetAssigned;
    }
    //Getters
    public String getClientName()
    {
        return clientName;
    }
        public int getBookingID()
    {
        return bookingID;
    }
    public int getArrivalDate()
    {
        return arrivalDate;
    }
    public int getAccomodationDays()
    {
        return accomodationDays;
    }
    public int getPeopleNum()
    {
        return peopleNum;
    }
    public Room getRoom()
    {
        return room;
    }
    
    public Booking(String clientName,int arrivalDate, int accomodationDays, int peopleNum) //Constructor για τα αντκείμενα τύπου κράτησης.
    {
        setBookingID();
        setClientName(clientName);
        setArrivalDate(arrivalDate);
        setAccomodationDays(accomodationDays);
        setPeopleNum(peopleNum);
        setRoom(null);
        System.out.println("Booking created");
    }
    
    public void printBooking()//Μεθοδος εκτύπωσης μίας κράτησης.
    {
        System.out.println("Client Name: " + getClientName() + "\nArrival Date: " + getArrivalDate() + "\nAccomodation Days: " + getAccomodationDays() + "\nPeopleNum: " + getPeopleNum() + "\nBookingID: " + getBookingID());
    }
}













